packages/asset-buyer/src/asset_buyer_manager.ts

```diff
@@ -0,0 +1,171 @@
+     *
+     * @return  An instance of AssetBuyerManager
+     */
+    public static getAssetBuyerManagerFromProvidedOrders(
+        provider: Provider,
+        orders: SignedOrder[],
+        feeOrders: SignedOrder[] = [],
+        options: Partial<AssetBuyerOpts>,
+    ): AssetBuyerManager {
+        const assetDatas = _.map(orders, order => order.makerAssetData);
+        const orderProvider = new BasicOrderProvider(_.concat(orders, feeOrders));
+        return new AssetBuyerManager(provider, assetDatas, orderProvider, options);
+    }
+    /**
+     * Instantiates a new AssetBuyerManager instance
+     * @param   provider                The Provider instance you would like to use for interacting with the Ethereum network.
+     * @param   assetDatas              The assetDatas of the desired assets to buy (for more info: https://github.com/0xProject/0x-protocol-specification/blob/master/v2/v2-specification.md).
+     * @param   orderProvider           An object that conforms to OrderProvider, see type for definition.
+     * @param   options                 Initialization options for an AssetBuyer. See type definition for details.
+     *
+     * @return  An instance of AssetBuyerManager
+     */
+    constructor(
+        provider: Provider,
+        assetDatas: string[],
+        orderProvider: OrderProvider,
+        options: Partial<AssetBuyerOpts>,
+    ) {
+        assert.assert(assetDatas.length > 0, `Expected 'assetDatas' to be a non-empty array.`);
+        this._assetBuyerMap = _.reduce(
+            assetDatas,
+            (accAssetBuyerMap: ObjectMap<AssetBuyer>, assetData: string) => {
+                accAssetBuyerMap[assetData] = new AssetBuyer(provider, assetData, orderProvider, options);
+                return accAssetBuyerMap;
+            },
+            {},
+        );
+    }
+    /**
+     * Get an AssetBuyer for the provided assetData
+     * @param   assetData   The desired assetData.
+     *
+     * @return  An instance of AssetBuyer
+     */
+    public getAssetBuyerFromAssetData(assetData: string): AssetBuyer {
+        const assetBuyer = this._assetBuyerMap[assetData];
+        if (_.isUndefined(assetBuyer)) {
+            throw new Error(`${AssetBuyerManagerError.AssetBuyerNotFound}: For assetData ${assetData}`);
+        }
+        return assetBuyer;
+    }
+    /**
+     * Get an AssetBuyer for the provided ERC20 tokenAddress
+     * @param   tokenAddress    The desired tokenAddress.
+     *
+     * @return  An instance of AssetBuyer
+     */
+    public getAssetBuyerFromERC20TokenAddress(tokenAddress: string): AssetBuyer {
+        const assetData = assetDataUtils.encodeERC20AssetData(tokenAddress);
+        return this.getAssetBuyerFromAssetData(assetData);
+    }
+    /**
+     * Get a list of all the assetDatas that the instance supports
+     *
+     * @return  An array of assetData strings
+     */
+    public getAssetDatas(): string[] {
```

Comment:
perhaps we want another public method like this: `doesSupportAssetData(assetData: string): bool`